app/Repositories/PostRepository.php

```diff
@@ -170,6 +170,54 @@ public function reviews($data)
     }
 
     /**
+     * Updates a post's tags when added or deleted.
+     *
+     * @param array $data
+     *
+     * @return
+     */
+    public function tag($data)
+    {
+        $post = Post::where(['id' => $data['post_id']])->first();
+
+        // Check if the post already has the tag.
+        // If so, soft delete. Otherwise, add the tag to the post.
+        if (in_array($data['tag_name'], $post->tagNames())) {
+            $post->untag($data['tag_name']);
+
+            // Create an event when the tag is deleted.
+            if (!in_array($data['tag_name'], $post->tagNames())) {
+                $post->tag_name = $data['tag_name'];
+                $post->admin_northstar_id = $data['admin_northstar_id'];
+                $post->action = 'delete';
+
+                $event = Event::create([
```

Comment:
OHHH this is brilliant, thanks @sbsmith86 ! Totally didn't even think to use `auth()->user()->northstar_id;` here and was getting hung up thinking that I'd have to pass this to the ModelServiceProvider somehow. I gotta think outside of the box more! Thank you for this - it works great!